Marlin/src/gcode/probe/M951.cpp

```diff
@@ -0,0 +1,41 @@
+/**
+ * Marlin 3D Printer Firmware
+ * Copyright (C) 2016 MarlinFirmware [https://github.com/MarlinFirmware/Marlin]
+ *
+ * Based on Sprinter and grbl.
+ * Copyright (C) 2011 Camiel Gubbels / Erik van der Zalm
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "../../inc/MarlinConfigPre.h"
+
+#if ENABLED(MAGNETIC_PARKING_EXTRUDER)
+
+#include "../gcode.h"
+#include "../../module/tool_change.h"
+
+void GcodeSuite::M951() {
+  if (parser.seenval('L')) parkingposx[0]         = parser.value_linear_units();
+  if (parser.seenval('R')) parkingposx[1]         = parser.value_linear_units();
+  if (parser.seenval('G')) parkinggrabdistance    = parser.value_linear_units();
+  if (parser.seenval('N')) parkingslowspeed       = parser.value_linear_units();
```

Comment:
The letters 'G' and 'N' are verboten as G-code parameters. You'll have to select something else.